### ISSUE TITLE
Race condition in fluentd tests

### ISSUE THREAD

**Issue by theganyo** [apigee/apigee-remote-service-golib#1]
There's a race condition in the TestFluend tests that occasionally causes the tests to fail. This is a test-only issue, does not affect production.

```
go test ./analytics/... -count=20 -timeout=5s -run "TestFluentd"
2020/06/10 15:23:21 INFO starting analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsMTLS630934084/temp
2020/06/10 15:23:21 INFO started analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsMTLS630934084/temp
2020/06/10 15:23:21 INFO closing analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsMTLS630934084/temp
2020/06/10 15:23:21 INFO closed analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsMTLS630934084/temp
2020/06/10 15:23:21 INFO starting analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsTLSSkipVerify549306390/temp
2020/06/10 15:23:21 INFO started analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsTLSSkipVerify549306390/temp
2020/06/10 15:23:21 INFO closing analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsTLSSkipVerify549306390/temp
2020/06/10 15:23:21 INFO closed analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsTLSSkipVerify549306390/temp
2020/06/10 15:23:21 INFO starting analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsNoTLS744475064/temp
2020/06/10 15:23:21 INFO started analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsNoTLS744475064/temp
2020/06/10 15:23:21 INFO closing analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsNoTLS744475064/temp
2020/06/10 15:23:21 INFO closed analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsNoTLS744475064/temp
2020/06/10 15:23:21 INFO starting analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsMTLS507604906/temp
2020/06/10 15:23:21 INFO started analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsMTLS507604906/temp
2020/06/10 15:23:21 INFO closing analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsMTLS507604906/temp
2020/06/10 15:23:21 INFO closed analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsMTLS507604906/temp
2020/06/10 15:23:21 INFO starting analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsTLSSkipVerify113172076/temp
2020/06/10 15:23:21 INFO started analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsTLSSkipVerify113172076/temp
2020/06/10 15:23:21 INFO closing analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsTLSSkipVerify113172076/temp
2020/06/10 15:23:21 ERROR dial: dial tcp 127.0.0.1:61735: connect: connection refused
2020/06/10 15:23:21 INFO closed analytics manager: /var/folders/hl/ktx057x160g3j5rfn_n78b4000d7w5/T/TestFluentdAnalyticsTLSSkipVerify113172076/temp
panic: test timed out after 5s

goroutine 73 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:1459 +0xdf
created by time.goFunc
	/usr/local/Cellar/go/1.14.3/libexec/src/time/sleep.go:168 +0x44

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000318120, 0x13c651e, 0x21, 0x13d7228, 0x1085401)
	/usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:1043 +0x37e
testing.runTests.func1(0xc0003a6000)
	/usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:1284 +0x78
testing.tRunner(0xc0003a6000, 0xc00005be10)
	/usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:991 +0xdc
testing.runTests(0xc00000e3a0, 0x168b8a0, 0x12, 0x12, 0x0)
	/usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:1282 +0x2a7
testing.(*M).Run(0xc00014c000, 0x0)
	/usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:1199 +0x15f
main.main()
	_testmain.go:78 +0x135

goroutine 62 [IO wait]:
internal/poll.runtime_pollWait(0x1d55ec8, 0x72, 0x0)
	/usr/local/Cellar/go/1.14.3/libexec/src/runtime/netpoll.go:203 +0x55
internal/poll.(*pollDesc).wait(0xc00014c998, 0x72, 0x0, 0x0, 0x13bbd41)
	/usr/local/Cellar/go/1.14.3/libexec/src/internal/poll/fd_poll_runtime.go:87 +0x45
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/Cellar/go/1.14.3/libexec/src/internal/poll/fd_poll_runtime.go:92
internal/poll.(*FD).Accept(0xc00014c980, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.14.3/libexec/src/internal/poll/fd_unix.go:384 +0x1d4
net.(*netFD).accept(0xc00014c980, 0x8, 0xc00026d4f8, 0x10e67e4)
	/usr/local/Cellar/go/1.14.3/libexec/src/net/fd_unix.go:238 +0x42
net.(*TCPListener).accept(0xc0001a0c00, 0x0, 0xc000318120, 0x17)
	/usr/local/Cellar/go/1.14.3/libexec/src/net/tcpsock_posix.go:139 +0x32
net.(*TCPListener).Accept(0xc0001a0c00, 0xc000318120, 0xc00001b540, 0x17, 0x3)
	/usr/local/Cellar/go/1.14.3/libexec/src/net/tcpsock.go:261 +0x64
crypto/tls.(*listener).Accept(0xc0001a0c20, 0x13bcf7a, 0xa, 0xc00026d6c0, 0x1)
	/usr/local/Cellar/go/1.14.3/libexec/src/crypto/tls/tls.go:55 +0x37
github.com/apigee/apigee-remote-service-golib/analytics.fluentdReceive(0xc000318120, 0xf127, 0xc000091760, 0x1, 0x1, 0x1378fa0, 0xc000331f20, 0x0, 0x0, 0x0, ...)
	/Users/theganyo/dev/apigee-remote-service-golib/analytics/fluentd_analytics_test.go:269 +0x2bd
github.com/apigee/apigee-remote-service-golib/analytics.TestFluentdAnalyticsTLSSkipVerify(0xc000318120)
	/Users/theganyo/dev/apigee-remote-service-golib/analytics/fluentd_analytics_test.go:140 +0x6b5
testing.tRunner(0xc000318120, 0x13d7228)
	/usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:991 +0xdc
created by testing.(*T).Run
	/usr/local/Cellar/go/1.14.3/libexec/src/testing/testing.go:1042 +0x357
FAIL	github.com/apigee/apigee-remote-service-golib/analytics	5.106s
FAIL
```

**Comment by rockspore** [comment]
I had a simple fix by breaking up the fluentdReceive() function into two - setting up a listener and receiving the record. Does this make sense?

**Comment by theganyo** [comment]
You bet! Thanks.

**Comment by rockspore** [comment]
Thanks! I created a pull request (having signed an individual SLA). Let me know if I did anything wrong.

**Comment by theganyo** [comment]
Looks great! I've merged. Thanks.
Fixed by #2
